### PR TITLE
Add macro option to stop and rerun actions

### PR DESCRIPTION
**Pull request by Destroy666x**
To close https://github.com/WarmUpTill/SceneSwitcher/issues/1055 May have missed something as the lag in big macro.cpp was unbearable, but works fine from what I see.

### CODE REVIEW

File: data/locale/en-US.ini [hunk 1]
```diff
@@ -140,6 +140,7 @@ AdvSceneSwitcher.macroTab.highlightPerformedActions="Highlight recently performe
 AdvSceneSwitcher.macroTab.newMacroRegisterHotkey="Register hotkeys to control the pause state of new macros"
 AdvSceneSwitcher.macroTab.currentDisableHotkeys="Register hotkeys to control the pause state of selected macro"
 AdvSceneSwitcher.macroTab.currentSkipExecutionOnStartup="Skip execution of actions of current macro on startup"
+AdvSceneSwitcher.macroTab.currentStopActionsIfNotDone="Stop and rerun actions if macro meets its conditions again before it finishes running, instead of skipping reexecution"
```
Comment (WarmUpTill) [inline data/locale/en-US.ini:143]: How about changing this to:
`Stop the actions of this macro, if they are running still, when a new execution is triggered.`

The conditions won't necessarily always be involved as the macro can also be started manually or via the "Macro" action.

File: lib/macro/macro.hpp [hunk 2]
```diff
@@ -168,6 +170,7 @@ class Macro {
 	bool _conditionSateChanged = false;
 	bool _performActionsOnChange = true;
 	bool _skipExecOnStart = false;
+    bool _stopActionsIfNotDone = false;
```
Comment (WarmUpTill) [inline lib/macro/macro.hpp:173]: Can you please rebase on the current master?
I think I finally fixed the issue that prevented clang-format form doing its job in the CI runs and thus spotting formatting issues.

**Comment by WarmUpTill** [comment]
Thanks! :)
Looks good in general.
I added a few comments where I think changes might be necessary.

File: data/locale/en-US.ini (hunk 1 above)
Comment (Destroy666x) [reply data/locale/en-US.ini:143]: Good point, should include rerun info though.

File: lib/macro/macro.hpp (hunk 2 above)
Comment (Destroy666x) [reply lib/macro/macro.hpp:173]: Sure

**Comment by Destroy666x** [comment]
I think something in the clang config needs to be changed too as VSCode doesn't auto do changes for me correctly anymore either. Unless it's due to the lag bug. I'll check some other day.

Anyways, why not autoformat instead of checking formatting? I could add pre-commit hook even if you'd like it to be optional in CI

**Comment by Destroy666x** [comment]
Is it on purpose that there are 2 of these now BTW?
image

**Comment by Destroy666x** [comment]
Probably version mismatch then as it doesn't really do anything special, it just runs the formatter. I'll check what the CI uses and will set the same to see if it fixes it.

**Comment by Destroy666x** [comment]
Ok, so the fallback arg is required for whatever reason. I assume that uses some default rules if they are not included in the file?

**Pull request merged**